### PR TITLE
Expose version via default name and ping responses (if DEVELOPER=1)

### PR DESCRIPTION
**Pull request by rustyrussell**
This is helpful for debugging, as many bug reports could be caused by the remote end being old.

### CODE REVIEW

File: common/ping.c [hunk 1]
```diff
@@ -53,3 +59,25 @@ u8 *make_ping(const tal_t *ctx, u16 num_pong_bytes, u16 padlen)
...
+	int i;
+
+	if (!fromwire_pong(pong, pong, &ignored))
+		return "Bad pong";
+
+	if (*num_pings_outstanding == 0)
+		return "Unexpected pong";
+
+	for (i = 0; i < tal_len(ignored); i++) {
+		if (ignored[i] < ' ' || ignored[i] == 127)
```
Comment (ZmnSCPxj) [inline common/ping.c]: Maybe also check for `ignored[i] >= 128`?  The other side may dump random data in the pong, that by chance may have a sequence of >=128 bytes at the start, but do not form valid UTF-8 data.  Our pong here is just the version, which is composed solely of ASCII characters, so restricting to ASCII should be OK.

File: lightningd/options.c [hunk 2]
```diff
@@ -610,16 +610,27 @@ void setup_color_and_alias(struct lightningd *ld)
...
+		char *name;
+		int taillen;
 
 		memcpy(&adjective, der+3, sizeof(adjective));
 		memcpy(&noun, der+3+sizeof(adjective), sizeof(noun));
 		noun %= ARRAY_SIZE(codename_noun);
 		adjective %= ARRAY_SIZE(codename_adjective);
+
+		/* Only use 32 characters */
+		name = tal_fmt(ld, "%s%s-",
```
Comment (ZmnSCPxj) [inline lightningd/options.c]: Suggest doing this only if DEVELOPER=1.

File: lightningd/options.c [hunk 3]
```diff
@@ -610,16 +610,27 @@ void setup_color_and_alias(struct lightningd *ld)
...
+			       codename_adjective[adjective],
+			       codename_noun[noun]);
+		assert(strlen(name) < 32);
+		taillen = 32 - strlen(name);
+		if (taillen > strlen(version()))
+			taillen = strlen(version());
+		tal_append_fmt(&name, version() + strlen(version()) - taillen);
```
Comment (ZmnSCPxj) [inline lightningd/options.c]: Missing `, "%s"`? Granted the version will not contain formatting characters etc. but the principle...

File: common/ping.c (hunk 1 above)
Comment (rustyrussell) [reply common/ping.c]: In future we might stuff more information into the pong response; I'm not so so worried about malformed UTF-8 as I am confusion in the logs...

File: lightningd/options.c (hunk 2 above)
Comment (rustyrussell) [reply lightningd/options.c]: You might be right; wasn't sure about this one.  Will change.

**Comment by cdecker** [comment]
ACK 583e5a7

**Pull request merged**